common/lib/profile/driver_configuration_profiles.ts

```diff
@@ -0,0 +1,438 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ 
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+ 
+  http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { ConfigurationProfile } from "./configuration_profile";
+import { ConfigurationProfilePresetCodes } from "./configuration_profile_codes";
+import { WrapperProperties } from "../wrapper_property";
+import { HostMonitoringPluginFactory } from "../plugins/efm/host_monitoring_plugin_factory";
+import { AuroraInitialConnectionStrategyFactory } from "../plugins/aurora_initial_connection_strategy_plugin_factory";
+import { AuroraConnectionTrackerPluginFactory } from "../plugins/connection_tracker/aurora_connection_tracker_plugin_factory";
+import { ReadWriteSplittingPluginFactory } from "../plugins/read_write_splitting_plugin_factory";
+import { FailoverPluginFactory } from "../plugins/failover/failover_plugin_factory";
+import { InternalPooledConnectionProvider } from "../internal_pooled_connection_provider";
+import { AwsPoolConfig } from "../aws_pool_config";
+import { StaleDnsPluginFactory } from "../plugins/stale_dns/stale_dns_plugin_factory";
+
+export class DriverConfigurationProfiles {
+  private static readonly MONITORING_CONNECTION_PREFIX = "monitoring-";
+  private static readonly activeProfiles: Map<string, ConfigurationProfile> = new Map<string, ConfigurationProfile>();
+  private static readonly presets: Map<string, ConfigurationProfile> = new Map<string, ConfigurationProfile>([
+    [
+      ConfigurationProfilePresetCodes.A0,
+      new ConfigurationProfile(
+        ConfigurationProfilePresetCodes.A0,
+        [],
+        new Map<string, any>([
+          //[WrapperProperties.CONNECT_TIMEOUT.name, 10000],
```

Comment:
Wondering at all the commented out lines?

Reply:
These commented properties are not yet available. They'll be implemented soon in a separate PR.